ICSharpCode.Decompiler/IL/Transforms/LocalFunctionDecompiler.cs

```diff
@@ -65,6 +65,8 @@ public void Run(ILFunction function, ILTransformContext context)
 		{
 			if (!context.Settings.LocalFunctions)
 				return;
+			if (IsLocalFunctionMethod(function.Method, context) || IsLocalFunctionDisplayClass(function.Method.ParentModule.PEFile, (TypeDefinitionHandle)function.Method.DeclaringTypeDefinition.MetadataToken, context))
+				return;
```

Comment:
Disable this when current function is local func or inside local func display class, since it's not always possible to restore nested local function struct in non-top scope.
It's not possible to restore the struct when current function is `Method2`.
```cs
void Method() {
	Method1<int>();
	void Method1<T1>() {
		int i = 0;
		Method2<int>();
		void Method2<T2>() {
			i = 0;
			Method1<int>();
		}
	}
}
```